llvm/lib/Target/X86/lab3/nogin_denis/X86NoginCountInstsPass.cpp

```diff
@@ -0,0 +1,47 @@
+#include "X86.h"
+#include "X86InstrInfo.h"
+#include "X86Subtarget.h"
+#include "llvm/CodeGen/MachineFunctionPass.h"
+#include "llvm/CodeGen/MachineInstrBuilder.h"
+
+using namespace llvm;
+
+class X86NoginCountInstsPass : public MachineFunctionPass {
+public:
+  static inline char ID = 0;
+
+  X86NoginCountInstsPass() : MachineFunctionPass(ID) {}
+
+  bool runOnMachineFunction(MachineFunction &MachFunc) override {
+    DebugLoc DebugLocation = MachFunc.front().begin()->getDebugLoc();
+    const TargetInstrInfo *InstrInfo = MachFunc.getSubtarget().getInstrInfo();
+    size_t Count = 0;
+
+    MachineBasicBlock *ReturnBlock = nullptr;
+
+    for (auto &BasicBlock : MachFunc) {
+      for (auto &Instr : BasicBlock) {
+        Count++;
+        if (Instr.isReturn()) {
+          ReturnBlock = &BasicBlock;
+        }
+      }
+    }
+
+    if (!ReturnBlock) {
+      ReturnBlock = &MachFunc.back();
+      DebugLocation = ReturnBlock->begin()->getDebugLoc();
+    }
+
+    BuildMI(*ReturnBlock, ReturnBlock->getFirstTerminator(), DebugLocation,
+            InstrInfo->get(X86::MOV64mr))
+        .addImm(Count)
```

Comment:
This is number of instructions in assembly file but not the number of really executed instructions during program run.

Reply:
I didn't manage to do it before 12 o'clock, I hope everything is fine

Reply:
Ok, lets think that you pushed last commit from London and in your timezone was 26th April yet.
But, tsss, don't tell anyone about that hack. :)